src/haywire/http_request.c

```diff
@@ -303,6 +328,7 @@ void hw_http_response_send(hw_http_response* response, void* user_data, http_res
     hw_string* response_buffer = create_response_buffer(response);
     
     write_context->connection = resp->connection;
+    write_context->request = resp->connection->request;
```

Comment:
Do we need to store this in `write_context`? Don't we have access to the current request via the connection? Oh wait. The problem is because someone could process the request in another thread and `connection->request` won't represent the proper request anymore?